python_modules/dagster-graphql/dagster_graphql/implementation/fetch_runs.py

```diff
@@ -450,9 +448,84 @@ def _fetch_runs_not_in_backfill(
     return runs[:limit]
 
 
+def _bulk_action_status_from_run_status(status: DagsterRunStatus) -> Sequence[BulkActionStatus]:
+    """Converts a DagsterRunStatus to the BulkActionStatuses that display as that DagsterRunStatus in the UI."""
+    if status == DagsterRunStatus.SUCCESS:
+        return [BulkActionStatus.COMPLETED_SUCCESS]
+    if status == DagsterRunStatus.FAILURE:
+        return [BulkActionStatus.FAILED, BulkActionStatus.COMPLETED_FAILED]
+    if status == DagsterRunStatus.CANCELED:
+        return [BulkActionStatus.CANCELED]
+    if status == DagsterRunStatus.CANCELING:
+        return [BulkActionStatus.CANCELING]
+    if status == DagsterRunStatus.STARTED:
+        return [BulkActionStatus.REQUESTED]
+
+    return []
+
+
+def _bulk_action_statuses_from_run_statuses(
+    statuses: Sequence[DagsterRunStatus],
+) -> Sequence[BulkActionStatus]:
+    full_list = []
+    for status in statuses:
+        full_list.extend(_bulk_action_status_from_run_status(status))
+
+    return full_list
+
+
+def _filters_apply_to_backfills(filters: RunsFilter) -> bool:
+    # the following filters do not apply to backfills, so skip fetching backfills if they are set
+    if (
+        (filters.run_ids is not None and len(filters.run_ids) > 0)
+        or filters.updated_after is not None
+        or filters.updated_before is not None
+        or filters.snapshot_id is not None
+    ):
+        return False
+    # if filtering by statuses and all are not valid backfill statuses, skip fetching backfills
+    if filters.statuses and len(_bulk_action_statuses_from_run_statuses(filters.statuses)) == 0:
+        return False
+
+    return True
+
+
+def _bulk_action_filters_from_run_filters(filters: RunsFilter) -> BulkActionsFilter:
+    converted_statuses = (
+        _bulk_action_statuses_from_run_statuses(filters.statuses) if filters.statuses else None
+    )
+    return BulkActionsFilter(
+        created_before=filters.created_before,
+        created_after=filters.created_after,
+        statuses=converted_statuses,
+        job_name=filters.job_name,
+        tags=filters.tags,
+    )
+
+
+def _replace_created_before_with_cursor(
+    filters: RunsFilter, created_before_cursor: Optional[datetime.datetime]
+):
+    """After the first page of results is returned, created_before_cursor will be less than
```

Comment:
We're sorting by run_id / backfill_id right?  Is there a reason why we need to change `created_before` here instead of adding both run_id / backfill_id into the cursor object?

Reply:
run_id and backfill_id are also in the cursor object. the created before part of the cursor object is to protect against an edge case where a user has only executed single runs and pages some number of pages into the runs feed. at this point the backfill cursor is still None. then a backfill is created. on the next page that backfill would show up because we dont have a backfill cursor to prevent it from appearing

Reply:
i'm not totally sure that answers your question, lmk